### PR TITLE
Asynchronous Export Progress Updates.

### PR DESCRIPTION
**Pull request by adunsulag**
This change adds progress indicators to the export process.  It will loop through each batch individually and process the batch while reporting progress in 5 second intervals on the currently processed batch.  If a batch fails it can be retried.

Added a new column to the task to hold the current result progress.

Changed up the index page to function as both the main index and an ajax controller.  Probably could break this apart but for expediency did it this way.

### CODE REVIEW

**Comment by adunsulag** [comment]
@sjpadgett Can you give this PR a go on your test databases and let me know if it is working well for you?  It appears to work well for me.

Here is some screenshots of the change.

image

image

File: interface/modules/custom_modules/oe-module-ehi-exporter/templates/oe-module-ehi-exporter/ehi-exporter-tasks.html.twig [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            </div>
+            <div class="card-body">
+                <h4><b>{{ "Patients to be exported:"|xlt }}</b> {{ task.getPatientIds()|length|text }}</h4>
+                <p>{{ "Export Task has been queued and will start shortly"|xlt }}</p>
+                <details class="mb-5">
+                    <summary>
+                        {{ "Expand for Export Zip Details"|xlt }}
+                    </summary>
+                    <p>{{ "Patient Pids To Be Exported"|xlt }}</p>
+                    <div class="patientPids">{{ task.getPatientIds()|json_encode }}</div>
```
Comment (bradymiller) [inline interface/modules/custom_modules/oe-module-ehi-exporter/templates/oe-module-ehi-exporter/ehi-exporter-tasks.html.twig:90]: would also put a |text here

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
more neat stuff!

2DV

File: interface/modules/custom_modules/oe-module-ehi-exporter/src/Services/EhiExporter.php [hunk 2]
```diff
@@ -72,6 +72,50 @@ public function __construct(private $modulePublicDir, private $modulePublicUrl,
         $this->cryptoGen = new CryptoGen();
     }
```
Comment (stephenwaite) [inline interface/modules/custom_modules/oe-module-ehi-exporter/src/Services/EhiExporter.php:74]: hi @adunsulag , just a minor php exception above
```
Deprecated: Creation of dynamic property OpenEMR\Modules\EhiExporter\Services\EhiExporter::$jobService is deprecated

Comment (sjpadgett) [reply interface/modules/custom_modules/oe-module-ehi-exporter/src/Services/EhiExporter.php:74]: What PHP version are you running @stephenwaite?

Comment (stephenwaite) [reply interface/modules/custom_modules/oe-module-ehi-exporter/src/Services/EhiExporter.php:74]: hi @sjpadgett, easy dev is on [8.2](https://php.watch/versions/8.2/dynamic-properties-deprecated)

**Comment by stephenwaite** [comment]
great work, congrats @adunsulag !
smaller

File: interface/modules/custom_modules/oe-module-ehi-exporter/src/Services/EhiExporter.php (hunk 2 above)
Comment (sjpadgett) [reply interface/modules/custom_modules/oe-module-ehi-exporter/src/Services/EhiExporter.php:74]: Thanks. I'm 8.1 guess I need to go ahead and upgrade to 8.3 for my windows dev but to be truthful, I'd rather not know about new errors!:)

**Review by bradymiller — APPROVED** [review]

**Pull request merged**